tests/visualization_tests/test_parallel_coordinate.py

```diff
@@ -413,3 +413,50 @@ def test_plot_parallel_coordinate_only_missing_params() -> None:
 
     figure = plot_parallel_coordinate(study)
     assert len(figure.data) == 0
+
+
+@pytest.mark.parametrize("value", [float("inf"), -float("inf"), float("nan")])
+def test_nonfinite_removed(value: float) -> None:
+
+    study = create_study()
+    study.add_trial(
+        create_trial(
+            value=0.0,
+            params={"param_a": 1e-6},
+            distributions={"param_a": FloatDistribution(0.0, 1.0)},
+        )
+    )
+    study.add_trial(
+        create_trial(
+            value=value,
+            params={"param_a": 0},
+            distributions={"param_a": FloatDistribution(0.0, 1.0)},
+        )
+    )
+
+    figure = plot_parallel_coordinate(study)
+    assert len(figure.data) == 1
```

Comment:
This assertion passes on current master. We shouldn't test if plot has data (since it always renders), but rather if the data is correct (no inf/nan on any axis).

Reply:
Done in [883fd83](https://github.com/optuna/optuna/pull/3471/commits/883fd83d7e2d4260e768bd1414308c5e65abcb47)